The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8307539220/job/22736852961) [(plain)](https://github.com/rust-lang/rust/commit/2cb109a4ba219f766e831ef37f9e185fd394fd2c/checks/22736852961/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=onur-ozkan
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_b7271b21-1a52-4755-bb35-efbce53910e5
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=86985a3de1c6beb77e9d8226cf75916cb50ad260
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_b7271b21-1a52-4755-bb35-efbce53910e5
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_b7271b21-1a52-4755-bb35-efbce53910e5
GITHUB_TRIGGERING_ACTOR=onur-ozkan
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/121543/merge
GITHUB_WORKFLOW_SHA=86985a3de1c6beb77e9d8226cf75916cb50ad260
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_20_X64=/opt/hostedtoolcache/go/1.20.14/x64
---
   Compiling libc v0.2.153
   Compiling memchr v2.5.0
   Compiling std v0.0.0 (/checkout/library/std)
   Compiling compiler_builtins v0.1.108
thread 'rustc' panicked at /checkout/compiler/rustc_middle/src/mir/interpret/allocation.rs:522:69:
called `Result::unwrap()` on an `Err` value: Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }
stack backtrace:
   0:     0x7ff3ed2ee8ce - <std[490127e40ed3d8f1]::sys_common::backtrace::_print::DisplayBacktrace as core[301f0576711e1b37]::fmt::Display>::fmt
   1:     0x7ff3ed347fef - core[301f0576711e1b37]::fmt::write
   2:     0x7ff3ed2e30e5 - <std[490127e40ed3d8f1]::sys::pal::unix::stdio::Stderr as std[490127e40ed3d8f1]::io::Write>::write_fmt
   3:     0x7ff3ed2ee6a4 - std[490127e40ed3d8f1]::sys_common::backtrace::print
   4:     0x7ff3ed2f18ca - std[490127e40ed3d8f1]::panicking::default_hook::{closure#1}
   5:     0x7ff3ed2f15a9 - std[490127e40ed3d8f1]::panicking::default_hook
   6:     0x7ff3ee07aed8 - <alloc[de649da472fd148b]::boxed::Box<rustc_driver_impl[ca2c613af98cc927]::install_ice_hook::{closure#0}> as core[301f0576711e1b37]::ops::function::Fn<(&dyn for<'a, 'b> core[301f0576711e1b37]::ops::function::Fn<(&'a core[301f0576711e1b37]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[301f0576711e1b37]::marker::Sync + core[301f0576711e1b37]::marker::Send, &core[301f0576711e1b37]::panic::panic_info::PanicInfo)>>::call
   7:     0x7ff3ed2f1fa8 - std[490127e40ed3d8f1]::panicking::rust_panic_with_hook
   8:     0x7ff3ed2f1d22 - std[490127e40ed3d8f1]::panicking::begin_panic_handler::{closure#0}
   9:     0x7ff3ed2eeea6 - std[490127e40ed3d8f1]::sys_common::backtrace::__rust_end_short_backtrace::<std[490127e40ed3d8f1]::panicking::begin_panic_handler::{closure#0}, !>
  11:     0x7ff3ed2a9bb5 - core[301f0576711e1b37]::panicking::panic_fmt
  12:     0x7ff3ed2aa233 - core[301f0576711e1b37]::result::unwrap_failed
  13:     0x7ff3eef248e3 - <rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::allocation::Allocation>::read_scalar::<rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt>
  13:     0x7ff3eef248e3 - <rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::allocation::Allocation>::read_scalar::<rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt>
  14:     0x7ff3eef6cf6f - <rustc_const_eval[94856c17c1f9cb5]::interpret::memory::AllocRef<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::pointer::CtfeProvenance, ()>>::read_scalar
  15:     0x7ff3eeec4ee0 - <rustc_const_eval[94856c17c1f9cb5]::interpret::eval_context::InterpCx<rustc_const_eval[94856c17c1f9cb5]::const_eval::machine::CompileTimeInterpreter>>::read_immediate_from_mplace_raw
  16:     0x7ff3eeec50f9 - <rustc_const_eval[94856c17c1f9cb5]::interpret::eval_context::InterpCx<rustc_const_eval[94856c17c1f9cb5]::const_eval::machine::CompileTimeInterpreter>>::read_immediate_raw::<rustc_const_eval[94856c17c1f9cb5]::interpret::operand::OpTy>
  17:     0x7ff3eefa5a6f - <rustc_const_eval[94856c17c1f9cb5]::interpret::validity::ValidityVisitor<rustc_const_eval[94856c17c1f9cb5]::const_eval::machine::CompileTimeInterpreter>>::read_immediate
  18:     0x7ff3eefa736c - <rustc_const_eval[94856c17c1f9cb5]::interpret::validity::ValidityVisitor<rustc_const_eval[94856c17c1f9cb5]::const_eval::machine::CompileTimeInterpreter>>::try_visit_primitive
  19:     0x7ff3eeee8e3b - <rustc_const_eval[94856c17c1f9cb5]::interpret::eval_context::InterpCx<rustc_const_eval[94856c17c1f9cb5]::const_eval::machine::CompileTimeInterpreter>>::validate_operand_internal
  20:     0x7ff3ef034eb1 - rustc_const_eval[94856c17c1f9cb5]::const_eval::eval_queries::eval_to_allocation_raw_provider
  21:     0x7ff3efd6c3e4 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 24usize]>>
  22:     0x7ff3eff3f2ea - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_middle[ba8fe9de73c6a0a0]::ty::ParamEnvAnd<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::GlobalId>)>>::call_once
  23:     0x7ff3f00f9352 - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::DefaultCache<rustc_middle[ba8fe9de73c6a0a0]::ty::ParamEnvAnd<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::GlobalId>, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 24usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  24:     0x7ff3effd9008 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  25:     0x7ff3ef01bd7d - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::DefaultCache<rustc_middle[ba8fe9de73c6a0a0]::ty::ParamEnvAnd<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::GlobalId>, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 24usize]>>>
  26:     0x7ff3ef02f5c6 - rustc_const_eval[94856c17c1f9cb5]::const_eval::eval_queries::eval_to_const_value_raw_provider
  27:     0x7ff3efd6ec04 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::eval_to_const_value_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 24usize]>>
  28:     0x7ff3eff40a9a - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::eval_to_const_value_raw::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_middle[ba8fe9de73c6a0a0]::ty::ParamEnvAnd<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::GlobalId>)>>::call_once
  29:     0x7ff3f00f9352 - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::DefaultCache<rustc_middle[ba8fe9de73c6a0a0]::ty::ParamEnvAnd<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::GlobalId>, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 24usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  30:     0x7ff3f00483d8 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::eval_to_const_value_raw::get_query_non_incr::__rust_end_short_backtrace
  31:     0x7ff3f11fcd2f - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::DefaultCache<rustc_middle[ba8fe9de73c6a0a0]::ty::ParamEnvAnd<rustc_middle[ba8fe9de73c6a0a0]::mir::interpret::GlobalId>, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 24usize]>>>
  32:     0x7ff3f11fecf0 - <rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt>::const_eval_global_id
  33:     0x7ff3f11fd95e - <rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt>::const_eval_poly
  34:     0x7ff3ef82e2a1 - <core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::filter_map::FilterMap<core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::enumerate::Enumerate<core[301f0576711e1b37]::slice::iter::Iter<rustc_middle[ba8fe9de73c6a0a0]::ty::VariantDef>>, <rustc_index[349b7f9aee06aea9]::slice::IndexSlice<rustc_target[22e2854f2c941197]::abi::VariantIdx, rustc_middle[ba8fe9de73c6a0a0]::ty::VariantDef>>::iter_enumerated::{closure#0}>, <rustc_middle[ba8fe9de73c6a0a0]::ty::adt::AdtDef>::discriminants::{closure#0}>, <rustc_mir_build[79901290b6f5e724]::build::Builder>::perform_test::{closure#1}>, <rustc_middle[ba8fe9de73c6a0a0]::mir::syntax::SwitchTargets>::new<core[301f0576711e1b37]::iter::adapters::filter_map::FilterMap<core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::enumerate::Enumerate<core[301f0576711e1b37]::slice::iter::Iter<rustc_middle[ba8fe9de73c6a0a0]::ty::VariantDef>>, <rustc_index[349b7f9aee06aea9]::slice::IndexSlice<rustc_target[22e2854f2c941197]::abi::VariantIdx, rustc_middle[ba8fe9de73c6a0a0]::ty::VariantDef>>::iter_enumerated::{closure#0}>, <rustc_middle[ba8fe9de73c6a0a0]::ty::adt::AdtDef>::discriminants::{closure#0}>, <rustc_mir_build[79901290b6f5e724]::build::Builder>::perform_test::{closure#1}>>::{closure#0}> as core[301f0576711e1b37]::iter::traits::iterator::Iterator>::fold::<(), <(_, _) as core[301f0576711e1b37]::iter::traits::collect::Extend<(_, _)>>::extend::extend<rustc_data_structures[5c90426e7edfd1e]::packed::Pu128, rustc_middle[ba8fe9de73c6a0a0]::mir::BasicBlock, smallvec[b4b46cc9b4c67c9c]::SmallVec<[rustc_data_structures[5c90426e7edfd1e]::packed::Pu128; 1usize]>, smallvec[b4b46cc9b4c67c9c]::SmallVec<[rustc_middle[ba8fe9de73c6a0a0]::mir::BasicBlock; 2usize]>>::{closure#0}>
  35:     0x7ff3ef73646a - <rustc_middle[ba8fe9de73c6a0a0]::mir::syntax::SwitchTargets>::new::<core[301f0576711e1b37]::iter::adapters::filter_map::FilterMap<core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::map::Map<core[301f0576711e1b37]::iter::adapters::enumerate::Enumerate<core[301f0576711e1b37]::slice::iter::Iter<rustc_middle[ba8fe9de73c6a0a0]::ty::VariantDef>>, <rustc_index[349b7f9aee06aea9]::slice::IndexSlice<rustc_target[22e2854f2c941197]::abi::VariantIdx, rustc_middle[ba8fe9de73c6a0a0]::ty::VariantDef>>::iter_enumerated::{closure#0}>, <rustc_middle[ba8fe9de73c6a0a0]::ty::adt::AdtDef>::discriminants::{closure#0}>, <rustc_mir_build[79901290b6f5e724]::build::Builder>::perform_test::{closure#1}>>
  36:     0x7ff3ef706857 - <rustc_mir_build[79901290b6f5e724]::build::Builder>::test_candidates
  37:     0x7ff3ef703707 - <rustc_mir_build[79901290b6f5e724]::build::Builder>::match_simplified_candidates
  38:     0x7ff3ef71a80a - <rustc_mir_build[79901290b6f5e724]::build::Builder>::match_candidates
  39:     0x7ff3ef6ff9dd - <rustc_mir_build[79901290b6f5e724]::build::Builder>::lower_match_tree
  40:     0x7ff3ef717c88 - <rustc_mir_build[79901290b6f5e724]::build::Builder>::match_expr
  41:     0x7ff3ef7132df - <rustc_mir_build[79901290b6f5e724]::build::Builder>::expr_into_dest
  42:     0x7ff3ef71b8db - <rustc_mir_build[79901290b6f5e724]::build::Builder>::in_scope::<<rustc_mir_build[79901290b6f5e724]::build::Builder>::expr_into_dest::{closure#0}::{closure#0}, ()>
  43:     0x7ff3ef716544 - <rustc_mir_build[79901290b6f5e724]::build::Builder>::expr_into_dest::{closure#0}
  44:     0x7ff3ef712bec - <rustc_mir_build[79901290b6f5e724]::build::Builder>::expr_into_dest
  45:     0x7ff3ef6e8859 - rustc_mir_build[79901290b6f5e724]::build::mir_build::{closure#0}
  46:     0x7ff3ef6e64bb - rustc_mir_build[79901290b6f5e724]::build::mir_built
  47:     0x7ff3efd7aed5 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_built::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
  48:     0x7ff3f0043a95 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_built::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::LocalDefId)>>::call_once
  49:     0x7ff3f0193c1e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  50:     0x7ff3f0016194 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_built::get_query_non_incr::__rust_end_short_backtrace
  51:     0x7ff3eebc6b3c - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>>
  52:     0x7ff3eebd9b52 - rustc_mir_transform[9cdeb1becfd488e5]::ffi_unwind_calls::has_ffi_unwind_calls
  53:     0x7ff3efd69805 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::has_ffi_unwind_calls::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 1usize]>>
  54:     0x7ff3efe143d5 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::has_ffi_unwind_calls::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::LocalDefId)>>::call_once
  55:     0x7ff3f018ab8e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  56:     0x7ff3eff1984b - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::has_ffi_unwind_calls::get_query_non_incr::__rust_end_short_backtrace
  57:     0x7ff3eebea961 - rustc_mir_transform[9cdeb1becfd488e5]::mir_const
  58:     0x7ff3efd7b0b5 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_const::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
  59:     0x7ff3efeb42a5 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_const::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::LocalDefId)>>::call_once
  60:     0x7ff3f0193c1e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  61:     0x7ff3f0016774 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_const::get_query_non_incr::__rust_end_short_backtrace
  62:     0x7ff3eebc6b3c - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>>
  63:     0x7ff3eebead34 - rustc_mir_transform[9cdeb1becfd488e5]::mir_promoted
  64:     0x7ff3efd58afa - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_promoted::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 16usize]>>
  65:     0x7ff3eff3369d - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_promoted::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::LocalDefId)>>::call_once
  66:     0x7ff3f0185bcf - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  67:     0x7ff3efe58d68 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_promoted::get_query_non_incr::__rust_end_short_backtrace
  68:     0x7ff3ef8cfd9a - rustc_borrowck[34c66346ee60cf35]::mir_borrowck
  69:     0x7ff3efd58715 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
  70:     0x7ff3f003e735 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::LocalDefId)>>::call_once
  71:     0x7ff3f0193c1e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  72:     0x7ff3f00196f4 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  73:     0x7ff3ef9e209c - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>>
  74:     0x7ff3efa0278c - <rustc_borrowck[34c66346ee60cf35]::type_check::TypeChecker>::prove_closure_bounds
  75:     0x7ff3efa1837d - <rustc_borrowck[34c66346ee60cf35]::type_check::TypeChecker>::check_rvalue
  76:     0x7ff3efa1e7a0 - <rustc_borrowck[34c66346ee60cf35]::type_check::TypeChecker>::typeck_mir
  77:     0x7ff3ef9f8775 - rustc_borrowck[34c66346ee60cf35]::type_check::type_check
  78:     0x7ff3ef93eab5 - rustc_borrowck[34c66346ee60cf35]::nll::compute_regions
  79:     0x7ff3ef8e7f10 - rustc_borrowck[34c66346ee60cf35]::do_mir_borrowck
  80:     0x7ff3ef8d00db - rustc_borrowck[34c66346ee60cf35]::mir_borrowck
  81:     0x7ff3efd58715 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
  82:     0x7ff3f003e735 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::LocalDefId)>>::call_once
  83:     0x7ff3f0193c1e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  84:     0x7ff3f00196f4 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  85:     0x7ff3ee8e0f5c - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::VecCache<rustc_span[e6ef414343d70c34]::def_id::LocalDefId, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>>
  86:     0x7ff3ee8f6a67 - rustc_hir_analysis[1af200eface9f448]::collect::type_of::opaque::find_opaque_ty_constraints_for_rpit
  87:     0x7ff3efd5c81a - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::type_of_opaque::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
  88:     0x7ff3efef8c59 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::type_of_opaque::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::DefId)>>::call_once
  89:     0x7ff3f00d9d8e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::DefIdCache<rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  90:     0x7ff3eff49747 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::type_of_opaque::get_query_non_incr::__rust_end_short_backtrace
  91:     0x7ff3ee9347ac - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::DefIdCache<rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>>
  92:     0x7ff3ee957ea0 - rustc_hir_analysis[1af200eface9f448]::collect::type_of::type_of
  93:     0x7ff3efd79337 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
  94:     0x7ff3eff0a639 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::type_of::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, rustc_span[e6ef414343d70c34]::def_id::DefId)>>::call_once
  95:     0x7ff3f00d9d8e - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::DefIdCache<rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
  96:     0x7ff3eff49147 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::type_of::get_query_non_incr::__rust_end_short_backtrace
  97:     0x7ff3ef59236c - rustc_middle[ba8fe9de73c6a0a0]::query::plumbing::query_get_at::<rustc_query_system[d75268e8c2c525f8]::query::caches::DefIdCache<rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>>
  98:     0x7ff3ef59ceeb - <rustc_privacy[99969b651919cb17]::ReachEverythingInTheInterfaceVisitor>::ty
  99:     0x7ff3ef59b45f - <rustc_privacy[99969b651919cb17]::EmbargoVisitor as rustc_hir[f34a12d6e74bf743]::intravisit::Visitor>::visit_item
 100:     0x7ff3ef5d4247 - <rustc_middle[ba8fe9de73c6a0a0]::hir::map::Map>::visit_all_item_likes_in_crate::<rustc_privacy[99969b651919cb17]::EmbargoVisitor>
 101:     0x7ff3ef5a1500 - rustc_privacy[99969b651919cb17]::effective_visibilities
 102:     0x7ff3efd6c1d3 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::effective_visibilities::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>
 103:     0x7ff3f0011f41 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::effective_visibilities::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, ())>>::call_once
 104:     0x7ff3f00e9cc3 - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::SingleCache<rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
 105:     0x7ff3effd9acb - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::effective_visibilities::get_query_non_incr::__rust_end_short_backtrace
 106:     0x7ff3ef5502e1 - rustc_passes[5c403fc422e0ffc2]::stability::check_unused_or_stable_features
 107:     0x7ff3ee2ad469 - <rustc_data_structures[5c90426e7edfd1e]::sync::parallel::ParallelGuard>::run::<(), rustc_interface[4e89be71bcc16113]::passes::analysis::{closure#0}::{closure#0}::{closure#2}>
 108:     0x7ff3ee2bbbe3 - <rustc_session[7ee5e9502ad422cc]::session::Session>::time::<(), rustc_interface[4e89be71bcc16113]::passes::analysis::{closure#0}>
 110:     0x7ff3efd79523 - rustc_query_impl[ec29ecd7cc2611bf]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ec29ecd7cc2611bf]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 1usize]>>
 111:     0x7ff3effd4451 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::analysis::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, ())>>::call_once
 111:     0x7ff3effd4451 - <rustc_query_impl[ec29ecd7cc2611bf]::query_impl::analysis::dynamic_query::{closure#2} as core[301f0576711e1b37]::ops::function::FnOnce<(rustc_middle[ba8fe9de73c6a0a0]::ty::context::TyCtxt, ())>>::call_once
 112:     0x7ff3f00e5a00 - rustc_query_system[d75268e8c2c525f8]::query::plumbing::try_execute_query::<rustc_query_impl[ec29ecd7cc2611bf]::DynamicConfig<rustc_query_system[d75268e8c2c525f8]::query::caches::SingleCache<rustc_middle[ba8fe9de73c6a0a0]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[ec29ecd7cc2611bf]::plumbing::QueryCtxt, false>
 113:     0x7ff3f0045581 - rustc_query_impl[ec29ecd7cc2611bf]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
 114:     0x7ff3ee028a9b - <rustc_middle[ba8fe9de73c6a0a0]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}::{closure#1}::{closure#3}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>
 115:     0x7ff3ee0878eb - <rustc_interface[4e89be71bcc16113]::interface::Compiler>::enter::<rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}::{closure#1}, core[301f0576711e1b37]::result::Result<core[301f0576711e1b37]::option::Option<rustc_interface[4e89be71bcc16113]::queries::Linker>, rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>
 116:     0x7ff3ee041364 - rustc_span[e6ef414343d70c34]::set_source_map::<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_interface[4e89be71bcc16113]::interface::run_compiler<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}>::{closure#0}::{closure#0}>
 117:     0x7ff3ee0b99c5 - <scoped_tls[2279e9aaa7811bc7]::ScopedKey<rustc_span[e6ef414343d70c34]::SessionGlobals>>::set::<rustc_interface[4e89be71bcc16113]::util::run_in_thread_pool_with_globals<rustc_interface[4e89be71bcc16113]::interface::run_compiler<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>
 118:     0x7ff3ee041891 - rustc_span[e6ef414343d70c34]::create_session_globals_then::<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_interface[4e89be71bcc16113]::util::run_in_thread_pool_with_globals<rustc_interface[4e89be71bcc16113]::interface::run_compiler<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#0}>
 119:     0x7ff3ee02b01e - std[490127e40ed3d8f1]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[4e89be71bcc16113]::util::run_in_thread_with_globals<rustc_interface[4e89be71bcc16113]::util::run_in_thread_pool_with_globals<rustc_interface[4e89be71bcc16113]::interface::run_compiler<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>
 120:     0x7ff3ee0c3491 - <<std[490127e40ed3d8f1]::thread::Builder>::spawn_unchecked_<rustc_interface[4e89be71bcc16113]::util::run_in_thread_with_globals<rustc_interface[4e89be71bcc16113]::util::run_in_thread_pool_with_globals<rustc_interface[4e89be71bcc16113]::interface::run_compiler<core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>, rustc_driver_impl[ca2c613af98cc927]::run_compiler::{closure#0}>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[301f0576711e1b37]::result::Result<(), rustc_span[e6ef414343d70c34]::ErrorGuaranteed>>::{closure#1} as core[301f0576711e1b37]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
 121:     0x7ff3ed2fd5de - <std[490127e40ed3d8f1]::sys::pal::unix::thread::Thread>::new::thread_start
 123:     0x7ff3ed15d850 - <unknown>
 124:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.
error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

warning: the ICE couldn't be written to `/checkout/rustc-ice-2024-03-16T12_04_31-9481.txt`: Read-only file system (os error 30)
note: rustc 1.78.0-nightly (86985a3de 2024-03-16) running on x86_64-unknown-linux-gnu


note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debug-assertions=on -Z unstable-options -C symbol-mangling-version=v0 -Z unstable-options -Z macro-backtrace -C split-debuginfo=off -C prefer-dynamic -Z inline-mir -C link-args=-Wl,-z,origin -C link-args=-Wl,-rpath,$ORIGIN/../lib -C embed-bitcode=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked
note: some of the compiler flags provided by cargo are hidden

query stack during panic:
query stack during panic:
#0 [eval_to_allocation_raw] const-evaluating + checking `cmp::Ordering::Less::{constant#0}`
#1 [eval_to_const_value_raw] simplifying constant for the type system `cmp::Ordering::Less::{constant#0}`
#2 [mir_built] building MIR for `iter::traits::iterator::Iterator::cmp_by::compare::{closure#0}`
#3 [has_ffi_unwind_calls] checking if `iter::traits::iterator::Iterator::cmp_by::compare::{closure#0}` contains FFI-unwind calls
#4 [mir_const] preparing `iter::traits::iterator::Iterator::cmp_by::compare::{closure#0}` for borrow checking
#5 [mir_promoted] promoting constants in MIR for `iter::traits::iterator::Iterator::cmp_by::compare::{closure#0}`
#6 [mir_borrowck] borrow-checking `iter::traits::iterator::Iterator::cmp_by::compare::{closure#0}`
#7 [mir_borrowck] borrow-checking `iter::traits::iterator::Iterator::cmp_by::compare`
#8 [type_of_opaque] computing type of opaque `iter::traits::iterator::Iterator::cmp_by::compare::{opaque#0}`
#9 [type_of] computing type of `iter::traits::iterator::Iterator::cmp_by::compare::{opaque#0}`
#10 [effective_visibilities] checking effective visibilities
end of query stack
end of query stack
note: no errors encountered even though delayed bugs were created

note: those delayed bugs will now be shown as internal compiler errors

error: internal compiler error: {OpaqueTypeKey { def_id: DefId(0:54807 ~ core[52bb]::iter::traits::iterator::Iterator::cmp_by::compare::{opaque#0}), args: [X/#0, Y/#1, F/#2] }: OpaqueTypeDecl { hidden_type: OpaqueHiddenType { span: library/core/src/iter/traits/iterator.rs:3667:13: 3670:14 (#0), ty: Closure(DefId(0:8329 ~ core[52bb]::iter::traits::iterator::Iterator::cmp_by::compare::{closure#0}), [X/#0, Y/#1, F/#2, i16, Binder(extern "RustCall" fn((X/#0, Y/#1)) -> ops::control_flow::ControlFlow<cmp::Ordering, ()>, []), (F/#2,)]) } }}
  = note: delayed at compiler/rustc_infer/src/infer/opaque_types/table.rs:43:43
  = note: delayed at compiler/rustc_infer/src/infer/opaque_types/table.rs:43:43
             0: <rustc_errors::DiagCtxtInner>::emit_diagnostic
             1: <rustc_errors::DiagCtxt>::emit_diagnostic
             2: <rustc_errors::diagnostic::Diag>::emit_producing_error_guaranteed
             3: <rustc_infer::infer::opaque_types::table::OpaqueTypeStorage as core::ops::drop::Drop>::drop
             5: rustc_borrowck::mir_borrowck
             5: rustc_borrowck::mir_borrowck
             6: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 8]>>
             7: <rustc_query_impl::query_impl::mir_borrowck::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId)>>::call_once
             8: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::VecCache<rustc_span::def_id::LocalDefId, rustc_middle::query::erase::Erased<[u8; 8]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
             9: rustc_query_impl::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
            10: rustc_middle::query::plumbing::query_get_at::<rustc_query_system::query::caches::VecCache<rustc_span::def_id::LocalDefId, rustc_middle::query::erase::Erased<[u8; 8]>>>
            11: rustc_hir_analysis::collect::type_of::opaque::find_opaque_ty_constraints_for_rpit
            12: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::type_of_opaque::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 8]>>
            13: <rustc_query_impl::query_impl::type_of_opaque::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId)>>::call_once
            14: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefIdCache<rustc_middle::query::erase::Erased<[u8; 8]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
            15: rustc_query_impl::query_impl::type_of_opaque::get_query_non_incr::__rust_end_short_backtrace
            16: rustc_middle::query::plumbing::query_get_at::<rustc_query_system::query::caches::DefIdCache<rustc_middle::query::erase::Erased<[u8; 8]>>>
            17: rustc_hir_analysis::collect::type_of::type_of
            18: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 8]>>
            19: <rustc_query_impl::query_impl::type_of::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId)>>::call_once
            20: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefIdCache<rustc_middle::query::erase::Erased<[u8; 8]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
            21: rustc_query_impl::query_impl::type_of::get_query_non_incr::__rust_end_short_backtrace
            22: rustc_middle::query::plumbing::query_get_at::<rustc_query_system::query::caches::DefIdCache<rustc_middle::query::erase::Erased<[u8; 8]>>>
            23: <rustc_privacy::ReachEverythingInTheInterfaceVisitor>::ty
            24: <rustc_privacy::EmbargoVisitor as rustc_hir::intravisit::Visitor>::visit_item
            25: <rustc_middle::hir::map::Map>::visit_all_item_likes_in_crate::<rustc_privacy::EmbargoVisitor>
            26: rustc_privacy::effective_visibilities
            27: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::effective_visibilities::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 8]>>
            28: <rustc_query_impl::query_impl::effective_visibilities::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, ())>>::call_once
            29: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 8]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
            30: rustc_query_impl::query_impl::effective_visibilities::get_query_non_incr::__rust_end_short_backtrace
            31: rustc_passes::stability::check_unused_or_stable_features
            32: <rustc_data_structures::sync::parallel::ParallelGuard>::run::<(), rustc_interface::passes::analysis::{closure#0}::{closure#0}::{closure#2}>
            34: rustc_interface::passes::analysis
            35: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 1]>>
            36: <rustc_query_impl::query_impl::analysis::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, ())>>::call_once
            37: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 1]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
---
          

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

warning: the ICE couldn't be written to `/checkout/rustc-ice-2024-03-16T12_04_31-9481.txt`: Read-only file system (os error 30)
note: rustc 1.78.0-nightly (86985a3de 2024-03-16) running on x86_64-unknown-linux-gnu


note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debug-assertions=on -Z unstable-options -C symbol-mangling-version=v0 -Z unstable-options -Z macro-backtrace -C split-debuginfo=off -C prefer-dynamic -Z inline-mir -C link-args=-Wl,-z,origin -C link-args=-Wl,-rpath,$ORIGIN/../lib -C embed-bitcode=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked
note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
end of query stack
thread 'rustc' panicked at library/core/src/panicking.rs:164:5:
panic in a destructor during cleanup

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

warning: the ICE couldn't be written to `/checkout/rustc-ice-2024-03-16T12_04_31-9481.txt`: Read-only file system (os error 30)
note: rustc 1.78.0-nightly (86985a3de 2024-03-16) running on x86_64-unknown-linux-gnu


note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debug-assertions=on -Z unstable-options -C symbol-mangling-version=v0 -Z unstable-options -Z macro-backtrace -C split-debuginfo=off -C prefer-dynamic -Z inline-mir -C link-args=-Wl,-z,origin -C link-args=-Wl,-rpath,$ORIGIN/../lib -C embed-bitcode=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked
note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
end of query stack
thread caused non-unwinding panic. aborting.
rustc exited with signal: 6 (SIGABRT) (core dumped)

Caused by:
Caused by:
  process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustc /checkout/obj/build/bootstrap/debug/rustc --crate-name core --edition=2021 library/core/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debug-assertions=on -Zunstable-options --check-cfg 'cfg(feature, values("debug_refcell", "panic_immediate_abort"))' -C metadata=70719e8645e6f000 -C extra-filename=-70719e8645e6f000 --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/release/deps -Csymbol-mangling-version=v0 '--check-cfg=cfg(feature,values(any()))' -Zunstable-options '--check-cfg=cfg(bootstrap)' '--check-cfg=cfg(stdarch_intel_sde)' '--check-cfg=cfg(no_fp_fmt_parse)' '--check-cfg=cfg(no_global_oom_handling)' '--check-cfg=cfg(no_rc)' '--check-cfg=cfg(no_sync)' '--check-cfg=cfg(netbsd10)' '--check-cfg=cfg(backtrace_in_libstd)' '--check-cfg=cfg(target_env,values("libnx","p2"))' '--check-cfg=cfg(target_arch,values("arm64ec","spirv","nvptx","xtensa"))' -Zmacro-backtrace -Csplit-debuginfo=off -Cprefer-dynamic -Zinline-mir -Clink-args=-Wl,-z,origin '-Clink-args=-Wl,-rpath,$ORIGIN/../lib' -Cembed-bitcode=yes '-Zcrate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/")' -Z binary-dep-depinfo` (exit status: 254)
Build completed unsuccessfully in 0:04:10
{"llvm-bitcode-linker":"test-fail"}
##[group]Building bootstrap
    Finished dev [unoptimized] target(s) in 0.03s
    Finished dev [unoptimized] target(s) in 0.03s
##[endgroup]
ERROR: Tool `book` was not recorded in tool state.
ERROR: Tool `nomicon` was not recorded in tool state.
ERROR: Tool `reference` was not recorded in tool state.
ERROR: Tool `rust-by-example` was not recorded in tool state.
ERROR: Tool `edition-guide` was not recorded in tool state.
ERROR: Tool `embedded-book` was not recorded in tool state.
  local time: Sat Mar 16 12:04:37 UTC 2024
  network time: Sat, 16 Mar 2024 12:04:37 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/bootstrap/src/core/build_steps/check.rs
+++ b/src/bootstrap/src/core/build_steps/check.rs
@@ -61,19 +61,42 @@ fn args(builder: &Builder<'_>) -> Vec<String> {
             }
         }
 
+        let all_args = std::env::args().collect::<Vec<_>>();
+
         args.extend(strings(&["--", "--cap-lints", "warn"]));
         args.extend(ignored_lints.iter().map(|lint| format!("-Aclippy::{}", lint)));
-        let mut clippy_lint_levels: Vec<String> = Vec::new();
-        allow.iter().for_each(|v| clippy_lint_levels.push(format!("-A{}", v)));
-        deny.iter().for_each(|v| clippy_lint_levels.push(format!("-D{}", v)));
-        warn.iter().for_each(|v| clippy_lint_levels.push(format!("-W{}", v)));
-        forbid.iter().for_each(|v| clippy_lint_levels.push(format!("-F{}", v)));
-        args.extend(clippy_lint_levels);
+        args.extend(get_clippy_rules_in_order(&all_args, allow, deny, warn, forbid));
         args.extend(builder.config.free_args.clone());
         args
     } else {
         builder.config.free_args.clone()
     }
+}
+
+/// We need to keep the order of the given clippy lint rules before passing them.
+/// Since clap doesn't offer any useful interface for this purpose out of the box,
+/// we have to handle it manually.
+pub(crate) fn get_clippy_rules_in_order(
+    all_args: &[String],
+    allow_rules: &[String],
+    deny_rules: &[String],
+    warn_rules: &[String],
+    forbid_rules: &[String],
+) -> Vec<String> {
+    let mut result = vec![];
+
+    for (prefix, item) in
+        [("-A", allow_rules), ("-D", deny_rules), ("-W", warn_rules), ("-F", forbid_rules)]
+    {
+        item.iter().for_each(|v| {
+            let rule = format!("{prefix}{v}");
+            let position = all_args.iter().position(|t| t == &rule).unwrap();
+            result.push((position, rule));
+        });
+    }
+
+    result.sort_by_key(|&(position, _)| position);
+    result.into_iter().map(|v| v.1).collect()
 }
 
 fn cargo_subcommand(kind: Kind) -> &'static str {

--- a/src/bootstrap/src/core/build_steps/check.rs
+++ b/src/bootstrap/src/core/build_steps/check.rs
@@ -64,8 +64,8 @@ fn args(builder: &Builder<'_>) -> Vec<String> {
         let all_args = std::env::args().collect::<Vec<_>>();
 
         args.extend(strings(&["--", "--cap-lints", "warn"]));
-        args.extend(ignored_lints.iter().map(|lint| format!("-Aclippy::{}", lint)));
         args.extend(get_clippy_rules_in_order(&all_args, allow, deny, warn, forbid));
+        args.extend(ignored_lints.iter().map(|lint| format!("-Aclippy::{}", lint)));
         args.extend(builder.config.free_args.clone());
         args
     } else {

--- a/src/bootstrap/src/core/config/tests.rs
+++ b/src/bootstrap/src/core/config/tests.rs
@@ -1,5 +1,6 @@
 use super::{flags::Flags, ChangeIdWrapper, Config};
 use crate::core::config::{LldMode, TomlConfig};
+use crate::core::build_steps::check::get_clippy_rules_in_order;
 
 use clap::CommandFactory;
 use serde::Deserialize;
@@ -11,12 +12,13 @@ use std::{
 };
 
 fn parse(config: &str) -> Config {
-    let config = format!("{config} \r\n build.rustc = \"/does-not-exists\" ");
     Config::parse_inner(
         &[
-            "check".to_owned(),
-            "--config=/does/not/exist".to_owned(),
-            "--skip-stage0-validation".to_owned(),
+            "check".to_string(),
+            "--set=build.rustc=/does/not/exist".to_string(),
+            "--set=build.cargo=/does/not/exist".to_string(),
+            "--config=/does/not/exist".to_string(),
+            "--skip-stage0-validation".to_string(),
         ],
         |&_| toml::from_str(&config).unwrap(),
     )
@@ -169,7 +171,10 @@ fn override_toml_duplicate() {
     Config::parse_inner(
         &[
             "check".to_owned(),
+            "--set=build.rustc=/does/not/exist".to_string(),
+            "--set=build.cargo=/does/not/exist".to_string(),
             "--config=/does/not/exist".to_owned(),
+            "--skip-stage0-validation".to_owned(),
             "--set=change-id=1".to_owned(),
             "--set=change-id=2".to_owned(),
         ],
@@ -192,7 +197,15 @@ fn profile_user_dist() {
             .and_then(|table: toml::Value| TomlConfig::deserialize(table))
             .unwrap()
     }
-    Config::parse_inner(&["check".to_owned()], get_toml);
+    Config::parse_inner(
+        &[
+            "check".to_owned(),
+            "--set=build.rustc=/does/not/exist".to_string(),
+            "--set=build.cargo=/does/not/exist".to_string(),
+            "--skip-stage0-validation".to_string(),
+        ],
+        get_toml,
+    );
 }
 
 #[test]
@@ -253,4 +266,35 @@ fn parse_change_id_with_unknown_field() {
 
     let change_id_wrapper: ChangeIdWrapper = toml::from_str(config).unwrap();
     assert_eq!(change_id_wrapper.inner, Some(3461));
+}
+
+#[test]
+fn order_of_clippy_rules() {
+    let args = vec![
+        "clippy".to_string(),
+        "--fix".to_string(),
+        "--allow-dirty".to_string(),
+        "--allow-staged".to_string(),
+        "-Aclippy:all".to_string(),
+        "-Wclippy::style".to_string(),
+        "-Aclippy::foo1".to_string(),
+        "-Aclippy::foo2".to_string(),
+    ];
+    let config = Config::parse(&args);
+
+    let actual = match &config.cmd {
+        crate::Subcommand::Clippy { allow, deny, warn, forbid, .. } => {
+            get_clippy_rules_in_order(&args, &allow, &deny, &warn, &forbid)
+        }
+        _ => panic!("invalid subcommand"),
+    };
+
+    let expected = vec![
+        "-Aclippy:all".to_string(),
+        "-Wclippy::style".to_string(),
+        "-Aclippy::foo1".to_string(),
+        "-Aclippy::foo2".to_string(),
+    ];
+
+    assert_eq!(expected, actual);
 }

